lib/logstash/outputs/elasticsearch/common.rb

```diff
@@ -128,16 +130,23 @@ def submit(actions)
         action = actions[idx]
         action_params = action[1]
 
+        # Retry logic: If it is success, we move on. If it is a failure, we have 3 paths:
+        # - For 409, we log and drop. there is nothing we can do
+        # - For a mapping error, we send to dead letter queue for a human to intervene at a later point.
+        # - For everything else there's mastercard. Yep, and we retry indefinitely. This should fix #572 and other transient network issues
         if SUCCESS_CODES.include?(status)
           next
         elsif CONFLICT_CODE == status && VERSION_TYPES_PERMITTING_CONFLICT.include?(action_params[:version_type])
           @logger.debug "Ignoring external version conflict: status[#{status}] failure[#{failure}] version[#{action_params[:version]}] version_type[#{action_params[:version_type]}]"
           next
-        elsif RETRYABLE_CODES.include?(status)
-          @logger.info "retrying failed action with response code: #{status} (#{failure})"
+        elsif DLQ_CODES.include?(status)
+          action_event = action[2]
+          @dlq_writer.write(event, "Could not index event to Elasticsearch. status: #{status}, action: #{action}, response: #{response}")
```

Comment:
Let's put a TODO item to change this api to be similar to our logging api where we can include context as an object:

Example proposed:

```
dlq.write(event, "Could not index event to Elasticsearch", status: status, action: action, response: response)
```